src/app/navigation/utils.ts

```diff
@@ -48,9 +50,23 @@ export const findRouteMatch = (routes: AnyRoute[], location: Location): AnyMatch
   }
 };
 
+export const matchSearch = (action: AnyMatch, search: string | undefined) => {
+  const previous = querystring.parse(search || '');
+
+  const route = querystring.parse(hasParams(action)
+    ? action.route.getSearch ? action.route.getSearch(action.params) : ''
+    : action.route.getSearch ? action.route.getSearch() : ''
+  );
+
+  return querystring.stringify({
+    ...previous,
+    ...route,
+  });
+};
+
 export const matchUrl = (action: AnyMatch) => hasParams(action)
   ? action.route.getUrl(action.params)
-  : action.route.getUrl();
+  : (action.route.getUrl as any)();
```

Comment:
Why this stopped working without `as any` and why did it work previously (it requires `params)`? :D